File: general_fizzbuzz.rb
#!/usr/bin/env ruby
# http://rosettacode.org/wiki/General_FizzBuzz

def general_fizzbuzz(max,f1,s1,f2,s2,f3,s3)
  (1..max).each { |i| p i % f3 == 0 ? s3 : i % f2 == 0 ? s2 : i % f1 == 0 ? s1 : i }
end

general_fizzbuzz(20,3,"Fizz",5,"Buzz",7,"Baxx")

File: general_fizzbuzz.swift
#!/usr/bin/env swift
// http://rosettacode.org/wiki/General_FizzBuzz

func generalFizzBuzz(max: Int, factorOne: (Int, String), factorTwo: (Int, String), factorThree: (Int,String)) {
  for i in 0...max {
    if i % factorThree.0 == 0 {
      println(factorThree.1)
    } else if i % factorTwo.0 == 0 {
      println(factorTwo.1)
    } else if i % factorOne.0 == 0 {
      println(factorOne.1)
    } else {
      println(i)
    }
  }
}

generalFizzBuzz(20,(3,"Fizz"),(5,"Buzz"),(7,"Baxx"))
